### ISSUE TITLE
New data object structure in Grafana 10.0.0

### ISSUE THREAD

**Issue by Rumbelstilzchen** [nline/nline-plotlyjs-panel#10]
Hello,
Within my docker test instance I tried Grafana 10.0.0 with plotly 1.4.0.
I liked to visualize a heatmap, but the plot is empty - only the grid and axes are visible.
With grafana up to 9.5.3 and plotly 1.3.0 the same panel and data worked as expected

**Comment by Meisterschueler** [comment]
I can confirm it is not working. I get an error message:
`TypeError: Cannot read properties of undefined (reading 'length') - line 4`

The first lines of my script is:
```
let dataset = [];

if (data.series.length > 0) {
  for (let i = 0; i < data.series[0].fields[0].values.buffer.length; i++) {
    let receiver = data.series[0].fields[0].values.buffer[i];
    let valstr = data.series[0].fields[1].values.buffer[i];
    let values = valstr.substr(1, valstr.length -2).split(',');
    let points = data.series[0].fields[2].values.buffer[i];
    let min_points = data.series[0].fields[3].values.buffer[i];
```

so for me it seems that buffer is null ...

**Comment by jacksongoode** [comment]
Grafana just came out a few days ago so there is some breakage likely. I'll fix this.

**Comment by jacksongoode** [comment]
Hi @Meisterschueler and @Rumbelstilzchen, just to follow up it seems like the way the values are stored has changed. You now access values directly from the `.values` property **without** the `.buffer`. This should allow your panels to continue to function.

```js
let x = data.series[0].fields[0].values;
let y = data.series[0].fields[1].values;
```

**Comment by Meisterschueler** [comment]
Thank you, I can confirm this is working for me.

**Comment by b-neufeld** [comment]
The column values fix is working for me, but I can't "see" column names any more. 

e.g. 

```if (data.series[6].fields[i].name == "my_column_name") {...}```

**Comment by jacksongoode** [comment]
Hmm, it looks like the `name` prop is still there? Unless your data source might have changed? You can always `console.log(data)` and inspect what props are available in your web console.

<img width="917" alt="Screen Shot 2023-06-21 at 11 57 39 AM" src="https://github.com/nline/nline-plotlyjs-panel/assets/54308792/e6508dab-34f9-4ea2-bfe0-a93c0eb0bad8">

**Comment by b-neufeld** [comment]
Thank you so much for the quick reply. The console tip was helpful in confirming all the data was in the right places Digging a little more, I found the issue. 

I had a bunch of variable assignments that were len(1) arrays, like this: 
```javascript 
let my_variable = data.series[0].fields[1].values.buffer[0]
```

Based on this thread, I changed them to:
```javascript 
let my_variable= data.series[0].fields[1].values
```

But then I had to change usage of my_variable in Plotly, from this:
```javascript 
  y: [my_variable],
```

To this (removed array brackets)
```javascript 
  y: my_variable,
```

The other thing that screwed me up, that I'll share in case someone else makes the same mistake... I did a find-and-replace for .buffer[0] with a blank string for a bunch of single assignments, e.g. change ```data.series[0].fields[1].values.buffer[0]``` to ```data.series[0].fields[1].values```. The correct form was ```data.series[0].fields[1].values```.
